### PR TITLE
feat: add error metrics to instance at start up to create a metric.

### PR DESCRIPTION
**Pull request by GuptaNavdeep1983**
This PR intends to add an error metric to CloudWatch in case of an error occurred during start-up script execution.

### CODE REVIEW

File: modules/runners/templates/start-runner.sh [hunk 1]
```diff
@@ -2,6 +2,34 @@
...
+  if [ "$agent_mode" = "ephemeral" ] || [ "$exit_code" -ne 0 ]; then
+    echo "Stopping CloudWatch service"
+    systemctl stop amazon-cloudwatch-agent.service || true
+    echo "Terminating instance"
+    aws ec2 terminate-instances \
+      --instance-ids "$instance_id" \
+      --region "$region" \
+      || true
+  fi
+}
```
Comment (npalm) [inline modules/runners/templates/start-runner.sh:29]: This part is now duplicated, or not. Maybe we can add a function terminate and invoke it on both places

**Review by npalm — CHANGES_REQUESTED** [review]

**Comment by npalm** [comment]
@GuptaNavdeep1983 did you had time for this PR?

**Comment by npalm** [comment]
@GuptaNavdeep1983 What should we do with this PR?

File: modules/runners/templates/start-runner.sh [hunk 2]
```diff
@@ -2,6 +2,34 @@
...
+cleanup() {
+  local exit_code="$1"
+  local error_location="$2"
+
+  if [ "$exit_code" -ne 0 ]; then
+    echo "Error $exit_code occurred on $error_location"
+    # Create a CloudWatch metric for error
+    aws cloudwatch put-metric-data \
+      --metric-name "RunnerInstanceUnhealthy" \
+      --namespace "Github Runners metrics" \
```
[the remark this answers is not in the document]
Comment (GuptaNavdeep1983) [reply modules/runners/templates/start-runner.sh:94]: This is not the health metric. Its created only on error which should be resolved immediately. I dont think it should be costly. I can remove the permissions (CloudWatch:PutMetricData) to put metric data conditionally  and let this call fail (with fail trap) in the absence of permissions. Do you find that okay?

**Pull request closed without merging**